ERCS/erc-draft_pm_capability.md

```diff
@@ -0,0 +1,285 @@
+---
+title: Paymaster Web Service Capability
+description: A standard for apps to communicate with smart wallets about paymaster web services
+author: Lukas Rosario (@lukasrosario)
+discussions-to: https://ethereum-magicians.org/t/add-erc-paymaster-web-service-capability/19530
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-04-03
+requires: 4337, 5792
+---
+
+## Abstract
+
+With [EIP-5792]('./eip-5792.md'), apps can communicate with wallets about advanced features via capabilities. This proposal defines a capability that allows apps to request that [ERC-4337]('./eip-4337.md') wallets communicate with a specified paymaster web service. To support this, we also define a standardized API for paymaster web services.
+
+## Motivation
+
+App developers want to start sponsoring their users' transactions using paymasters. Paymasters are commonly used via web services. However, there is currently no way for apps to tell wallets to communicate with a specific paymaster web service. Similarly, there is no standard for how wallets should communicate with these services. We need both a way for apps to tell wallets to communicate with a specific paymaster web service and a communication standard for wallets to do so.
+
+## Specification
+
+One new [EIP-5792]('eip-5792.md') wallet capability is defined. We also define a standard interface for paymaster web services as a prerequisite.
+
+### Paymaster Web Service Interface
+
+We define two JSON-RPC methods to be implemented by paymaster web services.
+
+#### `pm_getPaymasterStubData`
+
+Returns stub values to be used in paymaster-related fields of an unsigned user operation for gas estimation. Accepts an unsigned user operation, entrypoint address, chain id, and a context object. Paymaster service providers can define fields that app developers should use in the context object. 
+
+##### `pm_getPaymasterStubData` RPC Specification
+
+```typescript
+// [userOp, entryPoint, chainId, context]
+type GetPaymasterStubDataParams = [
```

Comment:
Not sure this needs to take any args? Aside maybe sender, but even then I am skeptical of entry point detection. I think maybe it should be passed. I expect most paymaster services have pretty static values here

Reply:
Oh I see paymaster address is passed

Reply:
Actually I take it back, it's probably useful to give the paymaster all of this incase they want to use in their dummy data.